### PR TITLE
NAV - add in standards and vocab to lesson materials page

### PR DESCRIPTION
**Pull request by kobryan0619**
Adds in vocab and standards rows with accurate links in data for english pages.  Open question noted in PR about handling this for other languages... 

Video of it working:

https://github.com/user-attachments/assets/a495753e-e3ce-4913-8aab-1d2ab74e013a

Screenshot from figma:

image

## Links

-  [Figma](https://www.figma.com/design/5ILfiJkPQpjskBwksMDeZl/Teacher-Dashboard?node-id=3148-1445&node-type=frame&m=dev)
- [Ticket](https://codedotorg.atlassian.net/jira/software/projects/TEACH/boards/65?selectedIssue=TEACH-1322)
- [Thread about how we are displaying these names here](https://codedotorg.slack.com/archives/C045UAX4WKH/p1727197144803619)

## Testing story

- Added to existing tests

### CODE REVIEW

File: dashboard/app/models/lesson.rb
Comment (kobryan0619) [inline dashboard/app/models/lesson.rb]: For both of these, I don't know how to confirm that these pages do indeed exist for the curriculum that the teacher is on NOR how to ensure that these are the right international urls. 

It looks like this is already inconsistent internationally - for example this course: https://studio.code.org/s/csd1-2021 has `https://studio.code.org/s/csd1-2021/normas` listed as the URL for standards when viewing the page in Spanish but that gives you a sad bee.  If you manually change this to `https://studio.code.org/s/csd1-2021/standards` you get the standards in Spanish...

Comment (kobryan0619) [reply dashboard/app/models/lesson.rb]: @mgc1194 do you have any insight on the localization question here?

File: apps/src/templates/teacherNavigation/ResourceRow.tsx
Comment (kobryan0619) [inline apps/src/templates/teacherNavigation/ResourceRow.tsx:12]: We don't need to provide the lessonNumber for the unit wide resources.  I made this an optional prop for this reason.

File: apps/src/templates/teacherNavigation/LessonMaterialsContainer.tsx
Comment (kobryan0619) [inline apps/src/templates/teacherNavigation/LessonMaterialsContainer.tsx:27]: You will see throughout here that I moved some of these props to optional props.  This felt like the cleaner code than passing in `null`.  the donwlodUrl in particular is used more for videos than other resources.

File: apps/src/templates/teacherNavigation/LessonMaterialsContainer.tsx
Comment (kobryan0619) [inline apps/src/templates/teacherNavigation/LessonMaterialsContainer.tsx]: I wanted to intentionally pass in `null` here since this is for `studentResources` which shouldn't be shown the standards or vocabulary. 

As you can see, I am kind of waffling between explicitly passing in something as `null` and creating optional props.  It is possible that this is another case where these should be optional and I should just check if they are undefined or not instead.  Interested in your thoughts, Dave.

File: apps/src/templates/teacherNavigation/LessonResources.tsx
Comment (kobryan0619) [inline apps/src/templates/teacherNavigation/LessonResources.tsx]: Checking for this makes sure that if we are rendering the `StudentResources` this content is not included in the table.

File: apps/src/templates/teacherNavigation/LessonMaterialsContainer.tsx
Comment (davidsbailey) [reply apps/src/templates/teacherNavigation/LessonMaterialsContainer.tsx]: good question! it would be slightly more conventional to make the prop optional. if there's a practical reason that passing null is preferred, I'm interested in hearing more about that (apologies I have not dug into the code review yet)

**Comment by MollyPeredo** [comment]
This looks great, Kaitie!

File: apps/src/templates/teacherNavigation/LessonResources.tsx
Comment (davidsbailey) [reply apps/src/templates/teacherNavigation/LessonResources.tsx]: why do we need to include `!lessonPlanUrl` in this conditional?

File: apps/src/templates/teacherNavigation/ResourceRow.tsx
Comment (davidsbailey) [inline apps/src/templates/teacherNavigation/ResourceRow.tsx:35]: per ongoing slack convo, unit numbers seem unreliable for CSD units with A and B options. since this feature is unlaunched, it seems ok to merge as-is if this PR becomes blocked on a decision on what to do here.

File: dashboard/app/models/lesson.rb
Comment (davidsbailey) [reply dashboard/app/models/lesson.rb]: we generally should not translate a url  part like `standards`. the link pointing to `normas` sounds like a bug, and the URL in your code seems like the right one to use. 

to get the correct url, you should use rails path helpers: https://guides.rubyonrails.org/routing.html#path-and-url-helpers in this case `standards_script_path`.

Comment (davidsbailey) [reply dashboard/app/models/lesson.rb]: I tried this for a course which has no lesson standards, and the page seems to render fine there: https://studio.code.org/s/mc/standards

one additional thought here is whether you'd want to exclude this entry for units with no standards. you can take a peek here to see how you might check whether there are any standards to show: https://github.com/code-dot-org/code-dot-org/blob/f29ccef4e4e7a1eba384fd952d32b747ddd64167/dashboard/app/controllers/scripts_controller.rb#L220-L221

File: dashboard/app/models/lesson.rb
Comment (davidsbailey) [inline dashboard/app/models/lesson.rb]: please use rails path helper. you shouldn't need to define the `unit_vocabulary_url` method (or the one above it), you can just use the rails path helper directly. here's how i got the path helper for standards. you should be able to repeat the process for vocab:
```
[development] dashboard > include Rails.application.routes.url_helpers
=> Object
[development] dashboard > Rails.application.routes.named_routes.helper_names.select {|name| name.include?('standard')}
=>
["standards_course_path",
 "search_standards_path",
 "standards_script_path",
 "dashboardapi_script_standards_path",
 "api_script_standards_path",
 "standards_course_url",
 "search_standards_url",
 "standards_script_url",
 "dashboardapi_script_standards_url",
 "api_script_standards_url"]
[development] dashboard > unit = Unit.find_by_name('csd3-2023')
=>
#<Unit:0x00007fdb202b8880
...
[development] dashboard > standards_script_path(unit)
=> "/s/csd3-2023/standards"
```

**Comment by davidsbailey** [comment]
I'm not quite sure how this fits in, but I'm not seeing anything on the lesson materials page for CSD Unit 6A, even though there are some standards:
Screenshot 2024-09-25 at 2 39 18 PM

I think it may be this early return: https://github.com/code-dot-org/code-dot-org/blob/1193373ed7bb4bb434041bdedc5bfc2426294ef8/apps/src/templates/teacherNavigation/LessonMaterialsContainer.tsx#L120-L122

maybe worth opening a Jira ticket for?

File: apps/src/templates/teacherNavigation/LessonResources.tsx
Comment (kobryan0619) [reply apps/src/templates/teacherNavigation/LessonResources.tsx]: In my mind, this was an additional check for "hey, this should only be included in the `Teacher Resoruces` area not the `Student Resources` area.  It was more of a "tool to communicate to a reader" than a "needed for logic tool".  As I write that, I think it makes more sense to just explicitly communicate that in a comment.  I am modifying that now.

File: apps/src/templates/teacherNavigation/ResourceRow.tsx
Comment (kobryan0619) [reply apps/src/templates/teacherNavigation/ResourceRow.tsx:35]: Added [this](https://codedotorg.atlassian.net/browse/TEACH-1378) ticket to cover this work.  This will also impact how we display other resources (currently those are also using "Unit Numbers" to indicate which lesson the resource belongs to.

File: apps/src/templates/teacherNavigation/LessonMaterialsContainer.tsx
Comment (kobryan0619) [reply apps/src/templates/teacherNavigation/LessonMaterialsContainer.tsx]: I ended up making this optional.

**Comment by kobryan0619** [comment]
> I'm not quite sure how this fits in, but I'm not seeing anything on the lesson materials page for CSD Unit 6A, even though there are some standards: Screenshot 2024-09-25 at 2 39 18 PM
> 
> I think it may be this early return:
> 
> https://github.com/code-dot-org/code-dot-org/blob/1193373ed7bb4bb434041bdedc5bfc2426294ef8/apps/src/templates/teacherNavigation/LessonMaterialsContainer.tsx#L120-L122
> 
> maybe worth opening a Jira ticket for?


Good catch!  I ended up somewhat fixing it, here is where it stands now:

image


But, I am guessing we will want to have some sort of design solution for if there are no student resources.  We do have a design solution if there is no lesson plan, but this is a bit different from that solution.   [Here's the ticket to address the student resources view.](https://codedotorg.atlassian.net/jira/software/projects/TEACH/boards/65?selectedIssue=TEACH-1379)

File: dashboard/app/models/lesson.rb
Comment (davidsbailey) [reply dashboard/app/models/lesson.rb]: thanks for switching to path helpers. here is a jira ticket to track the `normas` issue: https://codedotorg.atlassian.net/browse/TEACH-1380

**Review by davidsbailey — APPROVED** [review]
Looks great Kaitie, thanks for all the follow-ups!

**Pull request merged**
